File: services/lucy-dotnet/src/Lucy.Auth.Api/Entities/MentorApplication.cs
using Lucy.Shared.Constants;

namespace Lucy.Auth.Api.Entities;

public sealed class MentorApplication
{
    public string ApplicationId { get; set; } = default!;
    public string UserId { get; set; } = default!;
    public string? LanguageId { get; set; }
    public string? CertificateUrl { get; set; }
    public string Status { get; set; } = CommonStatus.Pending;
    public string? RejectReason { get; set; }
    public DateTime SubmittedAt { get; set; } = DateTime.UtcNow;

    public User User { get; set; } = null!;
    public Language? Language { get; set; }
}

File: lucy-web-stack/lucy-backend/src/main/java/com/lucy/backend/auth/entity/MentorApplication.java
package com.lucy.backend.auth.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Table(name = "MentorApplications")
@Data
public class MentorApplication {
    @Id
    @Column(name = "ApplicationId", length = 50)
    private String applicationId;

    @Column(name = "UserId", length = 50, nullable = false)
    private String userId;

    @Column(name = "LanguageId", length = 50)
    private String languageId;

    @Column(name = "CertificateUrl", length = 255)
    private String certificateUrl;

    @Column(name = "Status", length = 30, nullable = false)
    private String status = "PENDING";

    @Column(name = "RejectReason", length = 255)
    private String rejectReason;

    @Column(name = "SubmittedAt", nullable = false)
    private LocalDateTime submittedAt = LocalDateTime.now();
}
